src/size.c

```diff
@@ -4,17 +4,20 @@
 
 // [[ register() ]]
 r_obj* ffi_size(r_obj* x, r_obj* frame) {
-  struct vec_error_opts err = {
-    .p_arg = vec_args.x,
-    .call = { .x = frame, .env = r_null }
-  };
-  return r_len(vec_size_opts(x, &err));
+  struct r_lazy call = { .x = frame, .env = r_null };
+  return r_len(vec_size_params(x, vec_args.x, call));
 }
 
 r_ssize vec_size(r_obj* x) {
+  return vec_size_params(x, vec_args.x, lazy_calls.vec_size);
+}
+
+r_ssize vec_size_params(r_obj* x,
+                        struct vctrs_arg* p_arg,
+                        struct r_lazy call) {
   struct vec_error_opts err = {
-    .p_arg = vec_args.x,
-    .call = lazy_calls.vec_size
+    .p_arg = p_arg,
+    .call = call
   };
   return vec_size_opts(x, &err);
```

Comment:
I've introduced `vec_size_params()` here. It proved to be a rather useful variant of `vec_size()`. I know we mentioned `vec_size4()` being another option, but this just felt like the right name for this in the context of everything else around it in this file.

Reply:
I would use `vec_size_3()` since the goal is to get a feel for this naming scheme. I don't think we should mass rename until we're sure about the scheme, and using the new scheme only in blank files isn't practical. So we should at least use it for new functions.

In this case we might want to come up with a suffix that signals error context parameters. `vec_size_with_ctxt()`?

Reply:
I went with `vec_size_3()` to use the emerging naming scheme